### PR TITLE
changes for DTS-31261 (Backend changes)

### PR DESCRIPTION
**Pull request by deepaktandn**
changes for DTS-31261 (Backend changes ): As discussed with Brahmanand I have done the backend changes for Flow Efficiency and Lead Time filters.

### CODE REVIEW

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/FlowEfficiencyServiceImpl.java [hunk 1]
```diff
@@ -23,15 +23,10 @@
...
 import java.time.LocalDateTime;
 import java.time.ZoneId;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.LinkedHashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
+import java.util.*;
```
Comment (shunaray) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/FlowEfficiencyServiceImpl.java:26]: remove wildcard imports

**Review by shunaray — CHANGES_REQUESTED** [review]
use correct formatter whole file is shown changed.

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/LeadTimeServiceImpl.java [hunk 2]
```diff
@@ -31,10 +31,13 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 
+import com.publicissapient.kpidashboard.apis.model.*;
+import com.publicissapient.kpidashboard.common.model.jira.JiraIssue;
 import org.apache.commons.collections4.CollectionUtils;
 import org.apache.commons.lang3.ObjectUtils;
 import org.apache.commons.lang3.tuple.Pair;
 import org.bson.types.ObjectId;
+import org.hamcrest.collection.IsEmptyCollection;
```
Comment (shunaray) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/LeadTimeServiceImpl.java:40]: use  proper imports of collections

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/FlowEfficiencyServiceImpl.java [hunk 3]
```diff
@@ -23,15 +23,19 @@
...
+import java.util.List;
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.Collections;
 import java.util.Optional;
+import java.util.HashSet;
 import java.util.stream.Collectors;
 
+import com.publicissapient.kpidashboard.apis.model.*;
```
Comment (shunaray) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/FlowEfficiencyServiceImpl.java:38]: still wildcard imports !!

**Review by shunaray — CHANGES_REQUESTED** [review]
Reformat with correct file formatter, whole file is showing changed, can't be reviewed.

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/FlowEfficiencyServiceImpl.java [hunk 4]
```diff
@@ -71,380 +70,356 @@
...
+        List<String> xAxisRange = new ArrayList<>(rangeList);
+        Collections.reverse(xAxisRange);
+
+        Set<String> newfilters = new LinkedHashSet<>();
+
+        if(filters.contains(OVERALL)) {
+            filters.remove(OVERALL);
+            newfilters.add(OVERALL);
+            newfilters.addAll(filters);
+        }
```
Comment (kunkambl) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/FlowEfficiencyServiceImpl.java:188]: above code not needed and do those changes in setDataCountMap() method

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/FlowEfficiencyServiceImpl.java [hunk 5]
```diff
@@ -71,380 +70,356 @@
...
-	 */
-	@Override
-	public Integer calculateKPIMetrics(Map<String, Object> stringObjectMap) {
-		return null;
-	}
-
-	@Override
-	public Double calculateThresholdValue(FieldMapping fieldMapping) {
-		return calculateThresholdValue(fieldMapping.getThresholdValueKPI170(), KPICode.FLOW_EFFICIENCY.getKpiId());
-	}
```
Comment (kunkambl) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/FlowEfficiencyServiceImpl.java:448]: Fix file formatting for both the files

**Comment by brahmanand1** [comment]
not needed hence closing and deleting branch

**Pull request closed without merging**